src/into.js

```diff
@@ -0,0 +1,16 @@
+var _curry3 = require('./internal/_curry3');
+var _isTransformer = require('./internal/_isTransformer');
+var _reduce = require('./internal/_reduce');
+var _stepCat = require('./internal/_stepCat');
+
+module.exports = _curry3(function into(to, xf, ls) {
+    var acc, fn;
+    if (_isTransformer(to)) {
+        fn = to;
+        acc = to.init();
+    } else {
+        fn = _stepCat(to);
+        acc = to;
+    }
+    return _reduce(xf(fn), acc, ls);
```

Comment:
I'd prefer:

```javascript
return _isTransformer(to) ? _reduce(xf(to), to.init(), ls)
                          : _reduce(xf(_stepCat(to)), to, ls);
```

This is what I derive from the current code, but I'd rather not have to do the variable substitutions in my head. :)